### PR TITLE
Added Encryption for DataBase properties

### PR DESCRIPTION
**Pull request by arvindgh**

### CODE REVIEW

**Comment by ambarkhan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 5
           

Complexity increasing per file
==============================
- Ginger/Ginger/Environments/AppDataBasesPage.xaml.cs  10
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5006741&bid=16516198)

File: Ginger/GingerCore/Environments/Database.cs [hunk 1]
```diff
@@ -223,36 +223,54 @@ public static List<string> DbTypes
...
        
         public string GetConnectionString()
         {
             string connStr = null;
-            bool res;
-            res = false;
+            
+            //Decryption
+            string User = GetDecryptOrCalcluatedValue(UserCalculated);
+            string Pass = GetDecryptOrCalcluatedValue(PassCalculated);
```
Comment (ambarkhan) [inline Ginger/GingerCore/Environments/Database.cs]: Codacy Issue found: [Rename 'Pass' which hides the property with the same name.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5006741)

File: Ginger/GingerCore/Environments/Database.cs [hunk 2]
```diff
@@ -223,36 +223,54 @@ public static List<string> DbTypes
...
+            string TNS = GetDecryptOrCalcluatedValue(TNSCalculated);
```
Comment (ambarkhan) [inline Ginger/GingerCore/Environments/Database.cs]: Codacy Issue found: [Rename 'TNS' which hides the property with the same name.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5006741)

File: Ginger/GingerCore/Environments/Database.cs [hunk 3]
```diff
@@ -223,36 +223,54 @@ public static List<string> DbTypes
...
+            if (String.IsNullOrEmpty(ConnString) == false)
             {
-                connStr = ConnectionStringCalculated.Replace("{USER}", UserCalculated);
-     
-                String deCryptValue = EncryptionHandler.DecryptString(PassCalculated, ref res, false);
-                if (res == true)
-                    { connStr = connStr.Replace("{PASS}", deCryptValue); }
-                else
-                    { connStr = connStr.Replace("{PASS}", PassCalculated); }
+                connStr = ConnString.Replace("{USER}", User);
```
Comment (ambarkhan) [inline Ginger/GingerCore/Environments/Database.cs]: Codacy Issue found: [Remove this useless assignment to local variable 'connStr'.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5006741)

File: Ginger/GingerCore/Environments/Database.cs [hunk 4]
```diff
@@ -223,36 +223,54 @@ public static List<string> DbTypes
...
-            res = false;
+            
+            //Decryption
+            string User = GetDecryptOrCalcluatedValue(UserCalculated);
+            string Pass = GetDecryptOrCalcluatedValue(PassCalculated);
+            string TNS = GetDecryptOrCalcluatedValue(TNSCalculated);
+            string ConnString = GetDecryptOrCalcluatedValue(ConnectionStringCalculated);
 
-            if (String.IsNullOrEmpty(ConnectionStringCalculated) == false)
+            if (String.IsNullOrEmpty(ConnString) == false)
```
Comment (ambarkhan) [inline Ginger/GingerCore/Environments/Database.cs]: Codacy Issue found: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5006741)

File: Ginger/GingerCore/Environments/Database.cs [hunk 5]
```diff
@@ -223,36 +223,54 @@ public static List<string> DbTypes
...
 
        
         public string GetConnectionString()
         {
             string connStr = null;
-            bool res;
-            res = false;
+            
+            //Decryption
+            string User = GetDecryptOrCalcluatedValue(UserCalculated);
```
Comment (ambarkhan) [inline Ginger/GingerCore/Environments/Database.cs]: Codacy Issue found: [Rename 'User' which hides the property with the same name.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5006741)

**Comment by ambarkhan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 5
           

Complexity increasing per file
==============================
- Ginger/Ginger/Environments/AppDataBasesPage.xaml.cs  10
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5006741&bid=16516198)

File: Ginger/GingerCore/Environments/Database.cs (hunk 2 above)
Comment (ambarkhan) [inline Ginger/GingerCore/Environments/Database.cs]: Codacy Issue found: [Rename 'TNS' which hides the property with the same name.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5006741)

File: Ginger/GingerCore/Environments/Database.cs (hunk 5 above)
Comment (ambarkhan) [inline Ginger/GingerCore/Environments/Database.cs]: Codacy Issue found: [Rename 'User' which hides the property with the same name.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5006741)

File: Ginger/GingerCore/Environments/Database.cs (hunk 4 above)
Comment (ambarkhan) [inline Ginger/GingerCore/Environments/Database.cs]: Codacy Issue found: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5006741)

File: Ginger/GingerCore/Environments/Database.cs (hunk 3 above)
Comment (ambarkhan) [inline Ginger/GingerCore/Environments/Database.cs]: Codacy Issue found: [Remove this useless assignment to local variable 'connStr'.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5006741)

File: Ginger/GingerCore/Environments/Database.cs (hunk 1 above)
Comment (ambarkhan) [inline Ginger/GingerCore/Environments/Database.cs]: Codacy Issue found: [Rename 'Pass' which hides the property with the same name.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5006741)

**Comment by ambarkhan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 5
           

Complexity increasing per file
==============================
- Ginger/Ginger/Environments/AppDataBasesPage.xaml.cs  10
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5006741&bid=16516198)

File: Ginger/GingerCore/Environments/Database.cs (hunk 5 above)
Comment (ambarkhan) [inline Ginger/GingerCore/Environments/Database.cs]: Codacy Issue found: [Rename 'User' which hides the property with the same name.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5006741)

File: Ginger/GingerCore/Environments/Database.cs (hunk 3 above)
Comment (ambarkhan) [inline Ginger/GingerCore/Environments/Database.cs]: Codacy Issue found: [Remove this useless assignment to local variable 'connStr'.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5006741)

File: Ginger/GingerCore/Environments/Database.cs (hunk 4 above)
Comment (ambarkhan) [inline Ginger/GingerCore/Environments/Database.cs]: Codacy Issue found: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5006741)

File: Ginger/GingerCore/Environments/Database.cs (hunk 2 above)
Comment (ambarkhan) [inline Ginger/GingerCore/Environments/Database.cs]: Codacy Issue found: [Rename 'TNS' which hides the property with the same name.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5006741)

File: Ginger/GingerCore/Environments/Database.cs (hunk 1 above)
Comment (ambarkhan) [inline Ginger/GingerCore/Environments/Database.cs]: Codacy Issue found: [Rename 'Pass' which hides the property with the same name.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5006741)

**Comment by ambarkhan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 2
           

Complexity increasing per file
==============================
- Ginger/Ginger/Environments/AppDataBasesPage.xaml.cs  10
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5006741&bid=16516198)

File: Ginger/GingerCore/Environments/Database.cs [hunk 6]
```diff
@@ -223,36 +223,54 @@ public static List<string> DbTypes
...
+            if (!String.IsNullOrEmpty(ConnString))
             {
-                connStr = ConnectionStringCalculated.Replace("{USER}", UserCalculated);
-     
-                String deCryptValue = EncryptionHandler.DecryptString(PassCalculated, ref res, false);
-                if (res == true)
-                    { connStr = connStr.Replace("{PASS}", deCryptValue); }
-                else
-                    { connStr = connStr.Replace("{PASS}", PassCalculated); }
+                connStr = ConnString.Replace("{USER}", Userid);
```
Comment (ambarkhan) [inline Ginger/GingerCore/Environments/Database.cs]: Codacy Issue found: [Remove this useless assignment to local variable 'connStr'.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5006741)

File: Ginger/GingerCore/Environments/Database.cs [hunk 7]
```diff
@@ -263,24 +281,26 @@ public string GetConnectionString()
                 }
                 else if (DBType == eDBTypes.DB2)
                 {
-                    connStr = "Server=" + TNSCalculated + ";Database=" + Name + ";UID=" + UserCalculated + "PWD=" + deCryptValue;
+                    //TODO: need to confirm connection string works aftr UID => ';' is missing
+                    //standard connection string format for DB2 is Server=myAddress:myPortNumber;Database=myDataBase;UID=myUsername;PWD=myPassword;
```
Comment (ambarkhan) [inline Ginger/GingerCore/Environments/Database.cs]: Codacy Issue found: [Remove this commented out code.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5006741)

**Comment by arvindgh** [comment]
No need to encrypt

**Pull request closed without merging**